webpack/assets/javascripts/react_app/redux/API/APIMiddleware.js

```diff
@@ -4,8 +4,8 @@ import { get } from './APIRequest';
 export const APIMiddleware = store => next => action => {
   const { type, key, payload = {}, url, actionTypes = {} } = action;
   if (type === API_OPERATIONS.GET) {
-    get(payload, url, store, actionTypeGenerator(key, actionTypes));
+    return get(payload, url, store, actionTypeGenerator(key, actionTypes));
```

Comment:
I feel like this could cause problems since the return value of `get` would be a promise and not a plain action object.
I think it's maybe better if it's this instead:
```suggestion
    get(payload, url, store, actionTypeGenerator(key, actionTypes));
    return next(action);
```

Reply:
Fun Friday reading: some code examples from Redux docs that show that returning `next(action)` is definitely valid.  https://redux.js.org/advanced/middleware#seven-examples

Reply:
So this?

```javascript
if (type === API_OPERATIONS.GET) {
  get(payload, url, store, actionTypeGenerator(key, actionTypes));
}

return next(action);
```

Reply:
@waldenraines  I was thinking the `return next(action)` would be at the end of the `if` and `else` blocks.  But I guess that's equivalent.

Reply:
I am only worried we haven't sent the action to the processing by next middlewares. Now we do. Should we send it, or should we not? The action is already processed in the get case, right?

Reply:
Per our discussion on #theforeman-dev we decided that we should indeed send a `next(action)` so that the action is not swallowed.